### ISSUE TITLE
Group Buy order management displays incorrectly (kg) or not at all (items)

### ISSUE THREAD

**Issue by lauriewayne** [openfoodfoundation/openfoodnetwork#6874]
## Description
When managing orders that are set to "Group Buy", the hub manager needs to see the the minimum and maximum amount a customer would buy in order to put together an order. When the hub manager has group Buy orders and goes to bulk order management for group buys, the manager can't see it. This makes the Group Buy function essentially worthless for a hub manager putting together large orders.

Only metric or "items" have ever been available on this screen - imperial measures were never added to it. 

## Expected Behavior
The hub manager should be able to see the aggregate minimum and maximum order sizes so that they can manage orders to make complete cases (for instance)

## Actual Behaviour
If the product unit is kg, the display is garbled and illegible (overwrites other text?). If the product unit is items, no data is displayed. If the product unit is pounds. no data is displayed. 

## Steps to Reproduce

1. Edit a product and set the "group buy" switch on that product
![Screenshot 2021-02-14 at 8 09 22 PM - Display 2 (1)](https://user-images.githubusercontent.com/37803238/107904561-0a191b00-6f01-11eb-8988-2efb7c2f8334.jpg)

![Screenshot 2021-02-14 at 8 14 28 PM - Display 2](https://user-images.githubusercontent.com/37803238/107904730-73009300-6f01-11eb-8b17-f5fa738221ef.jpg)

2. Add product to order cycle, create order for product that is a min/max range
3. Close order cycle and go to admin -> orders -> bulk order management 
4. Click on product and tick the "shared resource" box (or don't, it doesn't make a difference)
If product is in kilograms, display for management/adjustment of quantities is not readable. If product unit is items, nothing is displayed to manage.

## Animated Gif/Screenshot
If product is in kilograms, display for management/adjustment of quantities is not readable:

![kg not display correctly](https://user-images.githubusercontent.com/37803238/107903895-40559b00-6eff-11eb-9bac-91d12a328117.jpg)

If product unit is items, nothing is displayed to manage:

![items no display](https://user-images.githubusercontent.com/37803238/107903974-6713d180-6eff-11eb-8266-47f5b101d26e.jpg)

## Workaround
Since neither kilograms nor items displays, there is no workaround. Ideally for this hub, items would be available as these are most relevant to them.

## Severity
bug-s2: a non-critical feature is broken, no workaround

## Your Environment
* Chrome on Chromebook and Android. Firefox on Windows 10
* Operating System and version (desktop or mobile):

**Comment by sigmundpetersen** [comment]
Am I correct in that this will be fixed by #6865 @filipefurtad0 @jibees ?

**Comment by filipefurtad0** [comment]
Yes, #6865 will addresses display issues, causing the overlapping text :+1: 

But it will not address other issues described in this issue, regarding that i) Items are not displaying any information, and that ii) imperial measures were not yet implemented.

So, maybe we keep this issue open and downgrade severity, after #6865 is merged? What do you think @sigmundpetersen @jibees ?

**Comment by jibees** [comment]
> we keep this issue open
And edit the description, taking into account that some part has already been addressed.

**Comment by RachL** [comment]
I'm afraid the remaining tasks are new features? In that case I think this should be open on discourse as wishlist and not github? Or did I miss something with the item part @filipefurtad0 ?

**Comment by lauriewayne** [comment]
According to @tschumilas  the "items" unit used to work so maybe this is a bug? I understand that imperial measures would be a new feature (and would not be a super high priority).

Dang it! Sorry I put two different issues in one!

**Comment by filipefurtad0** [comment]
@RachL @lauriewayne , I assumed - perhaps incorrectly... - that we would treat both cases (Items and imperial units not working) as bugs, since we support group buy for these two units, since PR #5888 (which introduced Imperial units).

I tried to observe reproduce this, but seems to work on a staging server for a **lb units** product - (sorry for the ugly overlap), but values seem to be displayed and updated:

![image](https://user-images.githubusercontent.com/49817236/107974469-8d447a80-6fae-11eb-9d0e-3085a158ca14.png)


![Peek 2021-02-15 16-41](https://user-images.githubusercontent.com/49817236/107973124-bb28bf80-6fac-11eb-9597-57bd4df01925.gif)

In the same staging server I did observe it not working for Items; so maybe this depends on other factors? I'm not sure, but there have been some reports of missing info from this page, like issue #6855. Maybe this issue is related?

**Comment by RachL** [comment]
@filipefurtad0 BOM was left out of the imperial unit work on purpose see https://community.openfoodnetwork.org/t/product-curation-meeting-25-notes-november-2nd-2020/2115/5?u=rachel. So it cannot be considered as a bug at this stage. As for item I wonder as well. Let's discuss it during delivery train tonight to be sure on the severity for items.

**Comment by filipefurtad0** [comment]
Thanks for pointing that out! :+1:

**Comment by tschumilas** [comment]
Very strange -  I have not been able to get the BOM of 'items' to work  with several different enterpises.  It is an established feature (described in the user guide) - so if its not working,  its a bug.   Let me try again and see if I can figure out other parameters that might be messing it up.  (Its been a few months since I've had a user ask for help with it)

**Comment by RachL** [comment]
Given the low number of people noticing this issue so far, I would be up to keep it as an s3, and if we get more feedback bump it as an s2.
What do others think?

**Comment by tschumilas** [comment]
SO just tested this again on OFN-CAN for 'item' type products -   result below.   Definately doesn't work.   Did the bulk buy work for you in your testing @filipefurtad0 ?    Here 2 hubs have asked about it -   but that was over a month ago and I haven't heard more grumbles.        Its an s2,   its described in the user guide,  and it doesn't work as per the user guide,  and it used to work (not sure when it stopped).   There is no workaround but its not a critical feature.  At least thats my thought.     Do we want to divide S2s into S2 with grumbling,  and S2 with no grumbling?  (only half joking -  you know what I'm saying - we have things to do  where there is a lot of grumbling....)
![bulk buy for 'item' type product](https://user-images.githubusercontent.com/13171778/108121391-bd545200-7070-11eb-8ba3-05a494ed077c.png)

**Comment by lauriewayne** [comment]
There is a large hub/buying club here (tens or hundreds of producers in their list) who is considering OFN and has been looking at it for a while, and this might be a showstopper for them. I can ask if it is (because walking away and signing up for another platform is a more common form of grumbling).

**Comment by sigmundpetersen** [comment]
Is there a chance that the item type not displaying could be caused by the display bug in #6833? Hidden in some way?
@filipefurtad0 @jibees has this case been tested on PR #6865? Is it worth staging this first and just testing this specific case to confirm?

**Comment by tschumilas** [comment]
Its a good thought @sigmundpetersen .   It works fine except that the numbers under the bulk buy headings aren't showing.  But user can adjust the quantities...    In fact,  at first I thought I had missed something in transifex....  but I don't think its that.

**Comment by filipefurtad0** [comment]
Yes, it could be that something is happening in the UI @sigmundpetersen  I just had a look by staging #6865. Below, **left side**, before fixing the UI, **right side** after fixing it.

I creating some products (Items, kg, lbs), allowing the bulk buy option and placing some orders. The video shows this the bulk order edit page for Items and Kg:

![Peek 2021-02-16 22-28](https://user-images.githubusercontent.com/49817236/108129594-594c8080-70a6-11eb-9d5f-401c7c31ca67.gif)

I would split this feature in two parts:
i) the board which is toggled, when clicking a given item: this should display information on the order, for that unit
ii) the arrows, which allow make changes to that order; showing the max. value for that order

For **Items**
i) the board is toggled, but **no info is displayed** on the buy headings -> I have never seen this info being displayed here (but maybe it was, at some point) 
ii) it's possible to make changes, and change the conversion factor to weight, for example, setting 1 Item to weight 500g; the max. value for that order is shown

For **Kg**
both i) and ii) work :+1:  (and I think for pounds, as well)

I understand what you mention about the user guide @tschumilas, users need to be able to rely on it. But in this case, I don't think the feature is really broken. I think making the changes works as expected, but the fact that no info is displayed, makes this more difficult. I'd rather say this is an S3, based on this. Am I missing an additional use of the numbers displayed on the board?

**Comment by sigmundpetersen** [comment]
From the user guide (https://guide.openfoodnetwork.org/basic-features/products-1/group-buy-for-bulk-ordering):

![Screenshot_20210217-000227_Firefox~2](https://user-images.githubusercontent.com/1451633/108132749-b3a00e00-70b3-11eb-8536-0ce6fe817f7a.jpg)

?

**Comment by tschumilas** [comment]
OMG -  is that a change to the guide?  I went through this feature in the user guide a month or so ago and I didn't see this notice - must have missed it.     I am sure that  bulk buy for items WAS working at one point in time -  I used it in my own food hub (but thats 2 years ago or more).   But if we don't consider it a currently supported feature (as in the current user guide description),  then I guess both the bulk buy per imperial weight,  and the bulk buy per item product types  are considered wishlist items.   I'm sorry -  if I had realized that we weren't supporting it right now,  I wouldn't have brought the issue this far.    Apologies for not seeing this in the user guide.

**Comment by filipefurtad0** [comment]
Interesting, thanks @sigmundpetersen - I totally missed that line as well...

So, yeah, let's close this @sigmundpetersen @RachL also as #6865 is now tested/merged, and will be on the next release.

**Comment by lauriewayne** [comment]
yes another great reason to

![image](https://user-images.githubusercontent.com/37803238/109407821-ea3e1a00-7938-11eb-8b0e-9974e4da6cfb.png)

**Comment by tschumilas** [comment]
So the issue:  apply bulk buy to 'item' type products -    needs to be a wishlist item.   @lauriewayne  -  I think it was one of your users that raised this -  do you want to create a wishlist item for it in Discourse?

**Comment by lauriewayne** [comment]
@tschumilas Yes I do! I have never done that before so I am excited to try!
